emcmake.py

```diff
@@ -41,6 +41,8 @@ def has_substr(args, substr):
   # On Windows specify MinGW Makefiles or ninja if we have them and no other
   # toolchain was specified, to keep CMake from pulling in a native Visual
   # Studio, or Unix Makefiles.
+  # Reduce cases of the concatented args e.g. "-GNinja" or "-GMingW Makefiles" by splitting them, fallback to default if none toolchain was specified.
+  args = [replaced for item in args for replaced in ([item[0:2], item[2:]] if (item[0:2] == "-G" and len(item) != 2) else [item])]
   if utils.WINDOWS and '-G' not in args:
```

Comment:
I think it would be better if we could fix this without modifying the args we pass through in the default case.

How about just changing `'-G' not in args` to `not any(a.startswith('-G') for a in args)`?   Basically looks for any arg that starts with `-G`?

Reply:
Other than that it would probably be good to add a test.   The tests for cmake live in test_other.py.  For example `test_cmake_install` which you can run using `./tests/runner other.test_cmake_install`.    For the new test you add you can do `if not WINDOWS: self.skipTest()` since this would be a windows-only test.

Reply:
Yes, its better. I'd change to that, and a test will be added.